e2e/stack.go

```diff
@@ -69,72 +64,69 @@ func New(
 }
 
 func (s *Stack) Run(ctx context.Context, env *environment.Env) error {
-	// Run anvil.
-	anvilCmd := exec.CommandContext( //nolint:gosec
-		ctx,
-		"anvil",
-		"--port", s.anvilURL.Port(),
-		"--order", "fifo",
-		"--disable-block-gas-limit",
-		"--gas-price", "0",
-		"--block-time", fmt.Sprint(s.l1BlockTime.Seconds()),
-	)
-	anvilCmd.Cancel = func() error {
-		// Anvil can catch SIGTERMs. The exec package sends a SIGKILL by default.
-		return anvilCmd.Process.Signal(syscall.SIGTERM)
+	// configure & run L1
+
+	const l2ChainID = 901
+	const networkName = "hardhat"
+	l1Deployments, err := opgenesis.NewL1Deployments("optimism/.devnet/addresses.json")
+	if err != nil {
+		return fmt.Errorf("new l1 deployments: %v", err)
 	}
-	if err := s.startCmd(anvilCmd); err != nil {
-		return err
+	deployConfig, err := opgenesis.NewDeployConfigWithNetwork(networkName, filepath.Join(s.contractsRootDir, "deploy-config"))
+	if err != nil {
+		return fmt.Errorf("new deploy config: %v", err)
 	}
-	env.Go(func() {
-		if err := anvilCmd.Wait(); err != nil && !errors.Is(err, ctx.Err()) {
-			s.eventListener.OnAnvilErr(fmt.Errorf("run %s: %v", anvilCmd, err))
-		}
-	})
-	// NOTE: should we set a timeout on the context? Might not be worth the complexity.
-	if !s.anvilURL.IsReachable(ctx) {
-		return nil
+	deployConfig.L2ChainID = l2ChainID // Ensure Monomer and the deploy config are aligned.
+
+	deployConfig.SetDeployments(l1Deployments)
+
+	// Generate a deployer key and pre-fund the account
+	deployerKey, err := crypto.GenerateKey()
+	if err != nil {
+		return fmt.Errorf("generate key: %v", err)
 	}
 
-	// Fund an account.
-	anvilRPCClient, err := rpc.DialContext(ctx, s.anvilURL.String())
+	var l1state state.Dump
+
+	l1StateJSON, err := os.ReadFile("optimism/.devnet/allocs-l1.json")
```

Comment:
nit: `filepath.Join` instead of raw string paths

Reply:
12d7acd